### PR TITLE
CA-162703: Don't log No_update exceptions from rrdd plugins

### PR DESCRIPTION
**Pull request by johnelse**

### CODE REVIEW

File: ocaml/rrdd/rrdd_server.ml [hunk 1]
```diff
@@ -628,12 +628,18 @@ module Plugin = struct
...
-			log_backtrace ();
+			let log e =
+				error "Failed to process plugin: %s (%s)" uid (Printexc.to_string e);
+				log_backtrace ()
+			in
 			match e with
-			| Invalid_header_string | Invalid_length | Invalid_checksum
-			| No_update as e -> raise e
-			| _ -> raise Read_error
+			| No_update -> raise No_update
```
Comment (euanh) [inline ocaml/rrdd/rrdd_server.ml:636]: Where is this exception eventually handled?    Will it bubble up to the handler in Rrdd_main.monitor_loop, which will log and restart?   I don't see any evidence of that happening in the logs, but I can't see anywhere else where exceptions from here should be handled.

Comment (johnelse) [inline ocaml/rrdd/rrdd_server.ml:636]: All exceptions get handled [here](https://github.com/xenserver/xen-api/blob/creedence-bugfix/ocaml/rrdd/rrdd_server.ml#L681-684) - if read_file raises anything then the mainloop will catch it silently and move on to the next plugin.

Comment (euanh) [inline ocaml/rrdd/rrdd_server.ml:636]: Aha, missed that.   I think the change is ok.

**Comment by johnelse** [comment]
Creedence port of https://github.com/xapi-project/xcp-rrdd/pull/30

**Pull request merged**